Data/IntSet/Internal.hs

```diff
@@ -661,102 +668,101 @@ isSubsetOf Nil _         = True
-split :: Key -> IntSet -> (IntSet,IntSet)
-split x t =
-  case t of
-      Bin _ m l r
-          | m < 0 -> if x >= 0  -- handle negative numbers.
-                     then case go x l of (lt :*: gt) -> let !lt' = union lt r
-                                                        in (lt', gt)
-                     else case go x r of (lt :*: gt) -> let !gt' = union gt l
-                                                        in (lt, gt')
-      _ -> case go x t of
-          (lt :*: gt) -> (lt, gt)
+split :: Key -> IntSet -> (IntSet, IntSet)
+split x t = case t of
+  Bin _ m l r | m < 0 ->
+    if x >= 0  -- handle negative numbers.
+    then case go x l of
+      (lt:*:gt) -> let !lt' = union lt r in (lt', gt)
+    else case go x r of
+      (lt:*:gt) -> let !gt' = union gt l in (lt, gt')
+  _ -> case go x t of
+    (lt:*:gt) -> (lt, gt)
   where
     go !x' t'@(Bin p m l r)
-        | match x' p m = if zero x' m
-                         then case go x' l of
-                             (lt :*: gt) -> lt :*: union gt r
-                         else case go x' r of
-                             (lt :*: gt) -> union lt l :*: gt
-        | otherwise   = if x' < p then (Nil :*: t')
-                        else (t' :*: Nil)
+      | match x' p m =
+        if zero x' m
+        then case go x' l of
+          (lt:*:gt) -> lt :*: union gt r
+        else case go x' r of
+          (lt:*:gt) -> union lt l :*: gt
+      | otherwise = if x' < p then (Nil :*: t') else (t' :*: Nil)
+
     go x' t'@(Tip kx' bm)
-        | kx' > x'          = (Nil :*: t')
-          -- equivalent to kx' > prefixOf x'
-        | kx' < prefixOf x' = (t' :*: Nil)
-        | otherwise = tip kx' (bm .&. lowerBitmap) :*: tip kx' (bm .&. higherBitmap)
-            where lowerBitmap = bitmapOf x' - 1
-                  higherBitmap = complement (lowerBitmap + bitmapOf x')
+      | kx' > x'          = (Nil :*: t')
+      | kx' < prefixOf x' = (t' :*: Nil) -- equivalent to kx' > prefixOf x'
+      | otherwise =
+        tip kx' (bm .&. lowerBitmap) :*: tip kx' (bm .&. higherBitmap)
+      where
+        lowerBitmap  = bitmapOf x' - 1
+        higherBitmap = complement (lowerBitmap + bitmapOf x')
     go _ Nil = (Nil :*: Nil)
 
 -- | /O(min(n,W))/. Performs a 'split' but also returns whether the pivot
 -- element was found in the original set.
-splitMember :: Key -> IntSet -> (IntSet,Bool,IntSet)
-splitMember x t =
-  case t of
-      Bin _ m l r | m < 0 -> if x >= 0
-                             then case go x l of
-                                 (lt, fnd, gt) -> let !lt' = union lt r
-                                                  in (lt', fnd, gt)
-                             else case go x r of
-                                 (lt, fnd, gt) -> let !gt' = union gt l
-                                                  in (lt, fnd, gt')
-      _ -> go x t
+splitMember :: Key -> IntSet -> (IntSet, Bool, IntSet)
+splitMember x t = case t of
+  Bin _ m l r | m < 0 ->
```

Comment:
This too looks weird.

Reply:
I think a hanging guard would help this since it would push the if/then/else further to the right separating it from the following `_ ->` case (which is the main cause of it looking weird imho).